src/uucore/src/lib/features/uptime.rs

```diff
@@ -368,20 +373,139 @@ pub fn get_loadavg() -> UResult<(f64, f64, f64)> {
 #[inline]
 pub fn get_formatted_loadavg() -> UResult<String> {
     let loadavg = get_loadavg()?;
-    Ok(format!(
-        "load average: {:.2}, {:.2}, {:.2}",
-        loadavg.0, loadavg.1, loadavg.2
+    Ok(get_message_with_args(
+        "uptime-lib-format-loadavg",
+        HashMap::from([
+            ("avg1".to_string(), format!("{:.2}", loadavg.0)),
+            ("avg5".to_string(), format!("{:.2}", loadavg.1)),
+            ("avg15".to_string(), format!("{:.2}", loadavg.2)),
+        ]),
     ))
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
-
+    use crate::locale;
+    use regex::Regex;
     #[test]
     fn test_format_nusers() {
+        unsafe {
+            std::env::set_var("LANG", "en_US.UTF-8");
+        }
+        let _ = locale::setup_localization("uptime");
         assert_eq!("0 users", format_nusers(0));
         assert_eq!("1 user", format_nusers(1));
         assert_eq!("2 users", format_nusers(2));
     }
+
+    #[test]
+    fn test_regex_with_new_format() {
+        use crate::locale::get_message_with_args;
+        use std::collections::HashMap;
+
+        // Force English locale and initialize localization
+        unsafe {
+            std::env::set_var("LANG", "en_US.UTF-8");
+        }
+        let _ = locale::setup_localization("uptime");
+
+        for test_days in [0, 1, 2] {
+            let output = get_message_with_args(
+                "uptime-format",
+                HashMap::from([
+                    ("days".to_string(), test_days.to_string()),
+                    ("hours".to_string(), " 0".to_string()),
+                    ("mins".to_string(), "05".to_string()),
+                ]),
+            );
+            println!("  {} days → '{}'", test_days, output);
+        }
+
+        // - Singular: "1 day, 2:05" (with comma)
+        // - Plural: "3 days 10:15" (without comma)
+        let regex =
+            Regex::new(r"up\s+(?:(\d+)\s+(?:day,\s+|days\s+))?(\d{1,2}):(\d{1,2})").unwrap();
+
+        let test_scenarios = [
+            ("0 days, 0 hours, 5 minutes", 0, 0, 5), // [0] case: should show no days
+            ("0 days, 2 hours, 30 minutes", 0, 2, 30), // [0] case: should show no days
+            ("1 day, 5 hours, 45 minutes", 1, 5, 45), // [one] case: singular
+            ("3 days, 10 hours, 15 minutes", 3, 10, 15), // [other] case: plural
+        ];
```

Comment:
I have to admit I'm a bit confused about this test. My expectation was, based on this setup, to see something like:
```rust
for (expected, days, hours, mins) in test_scenarios {
    assert_eq!(expected, call_some_function(days, hours, mins));
}
```